examples/dreambooth/train_dreambooth_flux.py

```diff
@@ -1553,41 +1586,21 @@ def get_sigmas(timesteps, n_dim=4, dtype=torch.float32):
                     guidance = None
 
                 # Predict the noise residual
-                if not args.train_text_encoder:
-                    model_pred = transformer(
-                        hidden_states=packed_noisy_model_input,
-                        # YiYi notes: divide it by 1000 for now because we scale it by 1000 in the transforme rmodel (we should not keep it but I want to keep the inputs same for the model for testing)
-                        timestep=timesteps / 1000,
-                        guidance=guidance,
-                        pooled_projections=pooled_prompt_embeds,
-                        encoder_hidden_states=prompt_embeds,
-                        txt_ids=text_ids,
-                        img_ids=latent_image_ids,
-                        return_dict=False,
-                    )[0]
-                else:
-                    prompt_embeds, pooled_prompt_embeds, text_ids = encode_prompt(
-                        text_encoders=[text_encoder_one, text_encoder_two],
-                        tokenizers=None,
-                        prompt=None,
-                        text_input_ids_list=[tokens_one, tokens_two],
-                    )
-                    model_pred = transformer(
-                        hidden_states=packed_noisy_model_input,
-                        # YiYi notes: divide it by 1000 for now because we scale it by 1000 in the transforme rmodel (we should not keep it but I want to keep the inputs same for the model for testing)
-                        timestep=timesteps / 1000,
-                        guidance=guidance,
-                        pooled_projections=pooled_prompt_embeds,
-                        encoder_hidden_states=prompt_embeds,
-                        txt_ids=text_ids,
-                        img_ids=latent_image_ids,
-                        return_dict=False,
-                    )[0]
-
+                model_pred = transformer(
+                    hidden_states=packed_noisy_model_input,
+                    # YiYi notes: divide it by 1000 for now because we scale it by 1000 in the transforme rmodel (we should not keep it but I want to keep the inputs same for the model for testing)
+                    timestep=timesteps / 1000,
+                    guidance=guidance,
+                    pooled_projections=pooled_prompt_embeds,
+                    encoder_hidden_states=prompt_embeds,
+                    txt_ids=text_ids,
+                    img_ids=latent_image_ids,
+                    return_dict=False,
+                )[0]
                 model_pred = FluxPipeline._unpack_latents(
                     model_pred,
-                    height=int(model_input.shape[2]),
-                    width=int(model_input.shape[3]),
+                    height=int(model_input.shape[2] * vae_scale_factor / 2),
+                    width=int(model_input.shape[3] * vae_scale_factor / 2),
```

Comment:
Where is this from?

https://github.com/huggingface/diffusers/blob/4e74206b0c443f9d272401f397d781d9d0630073/src/diffusers/pipelines/flux/pipeline_flux.py#L760

We don't additionally scale it by "/2".

Reply:
it's just a modification from the original version of the training scripts where we had
```
model_pred = FluxPipeline._unpack_latents(
                    model_pred,
                    height=int(model_input.shape[2] * 8),
                    width=int(model_input.shape[3] * 8),
                    vae_scale_factor=vae_scale_factor,
                )
```
and it didnt work with all resolutions, so we fixed in the previous PR for the LoRA script

(in the pipeline there is this- [diffusers/src/diffusers/pipelines/flux/pipeline_flux.py](https://github.com/huggingface/diffusers/blob/4e74206b0c443f9d272401f397d781d9d0630073/src/diffusers/pipelines/flux/pipeline_flux.py#L492))

Reply:
Yeah but we still don't have to do the additional scaling in the original pipeline, no? And it works with multiple resolutions without that. So, I am struggling to understand why we would need it here.

Reply:
in the pipeline we first scale in `prepare_latents` the width and height by 
```
height = 2 * (int(height) // self.vae_scale_factor)
width = 2 * (int(width) // self.vae_scale_factor)
```
but we don't override them, so when they're sent to `unpack_latents` it's the x8 of the scaled version here^
in the training script, we send
```
height=model_input.shape[2],
width=model_input.shape[3],
```
to `pack_latents` - which is already a scaled down version because it happens after vae encoding. 
i.e. `model_input.shape[2] is equivalent to 2 `* `(int(height) // self.vae_scale_factor)` in shape
that's why when we call `unpack_latents` in the training script we need to scale up

Reply:
Ah okay. Thanks for explaining this. Perhaps we could add a link to this comment in our script for our bookkeeping?

Reply:
done :)